ui/src/pages/RagChatTab/ChatOutput/Sources/SourceCard.tsx

```diff
@@ -129,8 +129,18 @@ export const SourceCard = ({ source }: { source: SourceNode }) => {
                 <Typography.Paragraph
                   style={{ textAlign: "left", whiteSpace: "pre-wrap" }}
                 >
-                  {chunkContents.data}
+                  {chunkContents.data?.text}
                 </Typography.Paragraph>
+                {chunkContents.data?.metadata.row_number ? (
+                  <>
+                    <Typography.Title level={5} style={{ marginTop: 0 }}>
+                      Metadata
+                    </Typography.Title>
+                    <Typography.Text>
+                      Row number: {chunkContents.data?.metadata.row_number}
```

Comment:
the check on line 134 means we know data exists.  

```suggestion
                      Row number: {chunkContents.data.metadata.row_number}
```